From 4ddeb68f35d0ba78a0d6f90ac32796812c265234 Mon Sep 17 00:00:00 2001
From: Jannick <thirdedition@gmx.net>
Date: Sun, 13 Jun 2021 22:18:23 +0200
Subject: [PATCH] tests(fix): handle YYINT dynamically determined at compile
 time

At compile time YYINT is determined to be one of 'short', 'int', 'long'
in 'defs.h'.

* test/run_test.sh(test_diffs):
  - add regular expression mapping 'typedef (long|short) YYINT;' to
    'typedef int YYINT;' used throughout the expected files
---
 test/run_test.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/run_test.sh b/test/run_test.sh
index 908a7d0..af39361 100755
--- a/test/run_test.sh
+++ b/test/run_test.sh
@@ -25,6 +25,7 @@ test_diffs() {
 			-e 's,#line \([1-9][0-9]*\) "'$REF_DIR'/,#line \1 ",' \
 			-e 's,#line \([1-9][0-9]*\) "'$TEST_DIR'/,#line \1 ",' \
 			-e 's,\(YACC:.* line [0-9][0-9]* of "\)'$TEST_DIR/',\1./,' \
+			-e 's/^typedef \(short\|long\) YYINT;$/typedef int YYINT;/' \
 			< $CMP >$tmpfile \
 			&& mv $tmpfile $CMP
 		if test ! -f $REF
-- 
2.32.0.windows.1

